services/controllerhost/event_handlers.go

```diff
@@ -1091,6 +1102,38 @@ func sealExtentOnStore(context *Context, storeUUID string, storeAddr string, ext
 	return err
 }
 
+// drainExtent sends a drain command to input host to gracefully
+// drain the clients connected to a given extent. If the input
+// host is not alive in ringpop, this method will return immediately
+func drainExtent(context *Context, dstID string, extentID string, inputID string) {
+	addr, err := context.rpm.ResolveUUID(common.InputServiceName, inputID)
+	if err != nil {
+		return
+	}
+	adminClient, err := common.CreateInputHostAdminClient(context.channel, addr)
+	if err != nil {
+		context.log.WithField(common.TagErr, err).Error(`drainExtent: failed to create input host client`)
+		return
+	}
+	drainExtentInfo := &admin.DrainExtents{
+		DestinationUUID: common.StringPtr(dstID),
+		ExtentUUID:      common.StringPtr(extentID),
+	}
+	drainReq := &admin.DrainExtentsRequest{
+		UpdateUUID: common.StringPtr(uuid.New()),
+		Extents:    []*admin.DrainExtents{drainExtentInfo},
+	}
+	context.log.WithFields(bark.Fields{
+		common.TagDst: common.FmtDst(dstID),
+		common.TagExt: common.FmtExt(extentID),
+		`reconfigID`:  drainReq.GetUpdateUUID(),
+	}).Info(`sending drain command to input host`)
+
+	ctx, cancel := thrift.NewContext(drainExtentTimeout)
+	adminClient.DrainExtent(ctx, drainReq)
```

Comment:
Won't it be better to catch the error here and record a log and/or a metric for tracking?